### PR TITLE
Mempool reorg covenants

### PR DESCRIPTION
**Pull request by pinheadmz**
Minimal alternative to https://github.com/handshake-org/hsd/pull/773

fixes the bugs without adding additional complexity.

### CODE REVIEW

File: lib/mempool/mempool.js [hunk 1]
```diff
@@ -193,13 +202,6 @@ class Mempool extends EventEmitter {
    */
 
   async _addBlock(block, txs, view) {
-    if (this.map.size === 0
```
Comment (nodech) [inline lib/mempool/mempool.js:196]: Why was this removed? (same as in another PR)

Comment (pinheadmz) [reply lib/mempool/mempool.js:196]: https://github.com/handshake-org/hsd/pull/773#discussion_r1062779385

**Review by nodech — APPROVED** [review]
